R/appMetadata-quarto.R

```diff
@@ -0,0 +1,66 @@
+inferQuartoInfo <- function(metadata, appDir, appPrimaryDoc) {
+  if (hasQuartoMetadata(metadata)) {
+    return(list(
+      version = metadata[["quarto_version"]],
+      engines = metadata[["quarto_engines"]]
+    ))
+  }
+
+  # If we don't yet have Quarto details, run quarto inspect ourselves
+  inspect <- quartoInspect(
+    appDir = appDir,
+    appPrimaryDoc = appPrimaryDoc
+  )
+  if (is.null(inspect)) {
+    return(NULL)
+  }
+
+  list(
+    version = inspect[["quarto"]][["version"]],
+    engines = I(inspect[["engines"]])
+  )
+}
+
+hasQuartoMetadata <- function(x) {
+  !is.null(x$quarto_version)
+}
+
+# Run "quarto inspect" on the target and returns its output as a parsed object.
+quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL) {
+  # If "quarto inspect appDir" fails, we will try "quarto inspect
+  # appPrimaryDoc", so that we can support single files as well as projects.
+  quarto <- quarto_path()
+  if (is.null(quarto)) {
+    abort("`quarto` is not installed")
```

Comment:
I assume that few people will hit this path because you'll usually have quarto installed if you're writing quarto files. But LMK if you have ideas for a more informative message.